src/localstore/incremental/JasmineGraphIncrementalLocalStore.cpp

```diff
@@ -11,31 +11,70 @@ See the License for the specific language governing permissions and
+#include <memory>
 #include <nlohmann/json.hpp>
+#include <stdexcept>
 using json = nlohmann::json;
 
 #include "../../util/logger/Logger.h"
+#include "../../centralstore/incremental/RelationBlock.h"
+
 
 Logger incremental_localstore_logger;
 
-JasmineGraphIncrementalLocalStore::JasmineGraphIncrementalLocalStore(unsigned int graphID, unsigned int partitionID){
+JasmineGraphIncrementalLocalStore::JasmineGraphIncrementalLocalStore(unsigned int graphID, unsigned int partitionID) {
     gc.graphID = graphID;
     gc.partitionID = partitionID;
-    gc.maxLabelSize = 10;
-    gc.openMode = "trunk";
+    gc.maxLabelSize = 43; // TODO tmkasun: read from .properties file
+    gc.openMode = "trunk"; // TODO tmkasun: read from .properties file
     this->nm = new NodeManager(gc);
 };
 
-std::string JasmineGraphIncrementalLocalStore::addEdgeFromString(std::string edgeString)
-{
+std::string JasmineGraphIncrementalLocalStore::addEdgeFromString(std::string edgeString) {
     try {
-        auto edge = json::parse(edgeString);
-        NodeBlock *sourceNode = this->nm->addNode(edge["id"]);
-    } catch (const std::exception &) {
+        auto edgeJson = json::parse(edgeString);
+
+        auto sourceJson = edgeJson["source"];
+        auto destinationJson = edgeJson["destination"];
+
+        std::string sId = std::string(sourceJson["id"]);
+        std::string dId = std::string(destinationJson["id"]);
+
+        RelationBlock* newRelation = this->nm->addEdge({sId, dId});
+        if (!newRelation) {
+            return "";
+        }
+        char value[PropertyLink::MAX_VALUE_SIZE] = {};
+
+        if (edgeJson.contains("properties")) {
+            auto edgeProperties = edgeJson["properties"];
+            for (auto it = edgeProperties.begin(); it != edgeProperties.end(); it++) {
+                strcpy(value, it.value().get<std::string>().c_str());
+                newRelation->addProperty(std::string(it.key()), &value[0]);
+            }
+        }
+
+        if (sourceJson.contains("properties")) {
+            auto sourceProps = sourceJson["properties"];
+            for (auto it = sourceProps.begin(); it != sourceProps.end(); it++) {
+                strcpy(value, it.value().get<std::string>().c_str());
+                newRelation->getSource()->addProperty(std::string(it.key()), &value[0]);
+            }
+        }
+        if (destinationJson.contains("properties")) {
+            auto destProps = destinationJson["properties"];
+            for (auto it = destProps.begin(); it != destProps.end(); it++) {
+                strcpy(value, it.value().get<std::string>().c_str());
+                newRelation->getDestination()->addProperty(std::string(it.key()), &value[0]);
+            }
+        }
+
+        incremental_localstore_logger.log("Added successfully!", "Info");
+    } catch (const std::exception&) { // TODO tmkasun: Handle multiple types of exceptions
+        incremental_localstore_logger.log("Erroneous edge data = " + edgeString, "error");
```

Comment:
Can we explain the reason why the data is erroneous? We should try to give more context into the error message.

Reply:
Updated the error message in this commit https://github.com/miyurud/jasminegraph/pull/132/commits/d8d887f5453408266df16065d675be6691184b72

Reply:
Ok